### ISSUE TITLE
`main` failing in parfors testing as of 6f70229

### ISSUE THREAD

**Issue by stuartarchibald** [numba/numba#8288]
This test (and others in the same `TestCase`) are failing: `FAIL: test_issue5942_1 (numba.tests.test_parfors.TestParforsMisc)` with:

<details>
<summary>Details:</summary>

```
======================================================================
FAIL: test_issue5942_1 (numba.tests.test_parfors.TestParforsMisc)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vsts/work/1/s/numba/tests/test_parfors.py", line 2988, in test_issue5942_1
    self.check(test_impl, gg, gg_next)
  File "/home/vsts/work/1/s/numba/tests/test_parfors.py", line 282, in check
    cfunc, cpfunc = self.compile_all(pyfunc, *args)
  File "/home/vsts/work/1/s/numba/tests/test_parfors.py", line 184, in compile_all
    cpfunc = self.compile_parallel(pyfunc, sig)
  File "/home/vsts/work/1/s/numba/tests/test_parfors.py", line 172, in compile_parallel
    return self._compile_this(func, sig, flags=self.pflags)
  File "/home/vsts/work/1/s/numba/tests/test_parfors.py", line 169, in _compile_this
    return compile_isolated(func, sig, flags=flags)
  File "/home/vsts/work/1/s/numba/core/compiler.py", line 309, in compile_isolated
    return compile_extra(typingctx, targetctx, func, args, return_type,
  File "/home/vsts/work/1/s/numba/core/compiler.py", line 716, in compile_extra
    return pipeline.compile_extra(func)
  File "/home/vsts/work/1/s/numba/core/compiler.py", line 452, in compile_extra
    return self._compile_bytecode()
  File "/home/vsts/work/1/s/numba/core/compiler.py", line 520, in _compile_bytecode
    return self._compile_core()
  File "/home/vsts/work/1/s/numba/core/compiler.py", line 495, in _compile_core
    raise e
  File "/home/vsts/work/1/s/numba/core/compiler.py", line 486, in _compile_core
    pm.run(self.state)
  File "/home/vsts/work/1/s/numba/core/compiler_machinery.py", line 364, in run
    raise e
  File "/home/vsts/work/1/s/numba/core/compiler_machinery.py", line 356, in run
    self._runPass(idx, pass_inst, state)
  File "/home/vsts/work/1/s/numba/core/compiler_lock.py", line 35, in _acquire_compile_lock
    return func(*args, **kwargs)
  File "/home/vsts/work/1/s/numba/core/compiler_machinery.py", line 311, in _runPass
    mutated |= check(pss.run_pass, internal_state)
  File "/home/vsts/work/1/s/numba/core/compiler_machinery.py", line 273, in check
    mangled = func(compiler_state)
  File "/home/vsts/work/1/s/numba/core/typed_passes.py", line 394, in run_pass
    self.typemap[var.name] = typ
  File "/home/vsts/work/1/s/numba/core/utils.py", line 375, in __setitem__
    raise AssertionError("key already in dictionary: %r" % (key,))
AssertionError: key already in dictionary: 'zero.2'
```

</details>

#8265 looks to be the cause. A small amount of debugging suggests that there are collisions in the typemap, this stems from scope confusion that lead to duplicate var names.

**Comment by esc** [comment]
@stuartarchibald can this be closed now that #8289 is merged?

**Comment by stuartarchibald** [comment]
> @stuartarchibald can this be closed now that #8289 is merged?

Yes! Closing!
